### PR TITLE
[telemetry] Add libc version to ray telemetry.

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?
This PR added GLIBC version to Ray telemetry. Linux distribution is not added because it needs distro pkg.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/usage/usage_lib.py [hunk 1]
```diff
@@ -356,6 +357,9 @@ def _generate_cluster_metadata():
                 "session_start_timestamp_ms": int(time.time() * 1000),
             }
         )
+        if sys.platform == "linux":
+            # Record glibc version
+            metadata.update({"glibc": platform.libc_ver()[1]})
```
Comment (jjyao) [inline python/ray/_private/usage/usage_lib.py:362]: You need to use `record_extra_usage_tag` otherwise we need to update the schema.

Comment (iycheng) [reply python/ray/_private/usage/usage_lib.py:362]: Which one do you think should be a system flag and which one should be extra usage tag?

Comment (jjyao) [reply python/ray/_private/usage/usage_lib.py:362]: Good question. We currently don't have a guideline for it. Currently extra usage tags are used by libraries to record their own usage stats so they are more like optional ones.

I think it's fine to be a system flag, I just need to update the schema and redeploy the server.

Comment (iycheng) [reply python/ray/_private/usage/usage_lib.py:362]: Let me know if it's hard and I can update it to use tags.
Should we make all of them as a flexible schema? I feel it's more convenient.

File: python/ray/tests/test_usage_stats.py [hunk 2]
```diff
@@ -1165,6 +1165,13 @@ def ready(self):
         assert payload["python_version"] == python_version
         assert payload["schema_version"] == "0.1"
         assert payload["os"] == sys.platform
+        if sys.platform == "linux":
```
Comment (rkooo567) [inline python/ray/tests/test_usage_stats.py:1168]: I am confused about the test. Isn't this supposed to be `sys.platform != "linux"`?

Comment (iycheng) [reply python/ray/tests/test_usage_stats.py:1168]: Good catch!

Comment (iycheng) [reply python/ray/tests/test_usage_stats.py:1168]: Good quesiton

**Comment by iycheng** [comment]
@rkooo567 You caught two actual issues! Good job! Thanks for helping reviewing this PR.

File: python/ray/tests/test_usage_stats.py [hunk 3]
```diff
@@ -1165,6 +1165,13 @@ def ready(self):
         assert payload["python_version"] == python_version
         assert payload["schema_version"] == "0.1"
         assert payload["os"] == sys.platform
+        if sys.platform != "linux":
+            assert "glibc" not in payload
+        else:
+            import platform
+
+            assert payload["glibc"] == platform.libc_ver()[1]
```
Comment (jjyao) [inline python/ray/tests/test_usage_stats.py:1173]: I think you also need to update the `schema` in the beginning of test_usage_stats.py

**Comment by iycheng** [comment]
@jjyao @rkooo567  any further comments for this PR?

**Comment by jjyao** [comment]
Let me take a look and also update the server schema to include `glibc_version`

File: python/ray/_private/usage/usage_lib.py [hunk 4]
```diff
@@ -759,6 +768,7 @@ def generate_report_data(
         extra_usage_tags=get_extra_usage_tags_to_report(gcs_client),
         total_num_nodes=get_total_num_nodes_to_report(gcs_client),
         total_num_running_jobs=get_total_num_running_jobs_to_report(gcs_client),
+        glibc_version=cluster_metadata.get("glibc_version"),
```
Comment (jjyao) [inline python/ray/_private/usage/usage_lib.py:771]: Set default value to ""?

File: python/ray/_private/usage/usage_lib.py [hunk 5]
```diff
@@ -356,6 +359,12 @@ def _generate_cluster_metadata():
                 "session_start_timestamp_ms": int(time.time() * 1000),
             }
         )
+        if sys.platform == "linux":
+            # Record glibc version
+            (lib, ver) = platform.libc_ver()
+            if lib != "glibc":
+                ver = "na"
```
Comment (jjyao) [inline python/ray/_private/usage/usage_lib.py:366]: nit: empty string?

**Review by jjyao — COMMENTED** [review]
Please get @thomasdesr approval as well

File: python/ray/_private/usage/usage_lib.py (hunk 4 above)
Comment (iycheng) [reply python/ray/_private/usage/usage_lib.py:771]: I think the type is: `        "glibc_version": {"type": ["null", "string"]},`, so shouldn't it be `None`?

File: python/ray/_private/usage/usage_lib.py (hunk 5 above)
Comment (iycheng) [reply python/ray/_private/usage/usage_lib.py:366]: Any reason for that? I'd prefer `na` if not fetched successfully.

If it's mac or windows, it'll be None.

File: python/ray/_private/usage/usage_lib.py (hunk 4 above)
Comment (jjyao) [reply python/ray/_private/usage/usage_lib.py:771]: Oh, I see. It will be converted to json null.

**Review by jjyao — APPROVED** [review]
Please also double check locally to make sure it works.

**Comment by iycheng** [comment]
@jjyao I think I need to change glibc version to libc version. According to feedback from @thomasdesr  this is more general.

**Review by rkooo567 — APPROVED** [review]

**Comment by iycheng** [comment]
@jjyao I feel we might need to change the schema again. Is this doable?

**Pull request merged**